changes-validation/src/main/scala/com/normation/plugins/changesvalidation/WorkflowService.scala

```diff
@@ -0,0 +1,348 @@
+
+  def startWorkflow(changeRequestId: ChangeRequestId, actor:EventActor, reason: Option[String]) : Box[WorkflowNodeId] =
+    if(cond().getOrElse(false)) whenTrue.startWorkflow(changeRequestId, actor, reason) else whenFalse.startWorkflow(changeRequestId, actor, reason)
+  def openSteps :List[WorkflowNodeId] =
+    if(cond().getOrElse(false)) whenTrue.openSteps else whenFalse.openSteps
+  def closedSteps :List[WorkflowNodeId] =
+    if(cond().getOrElse(false)) whenTrue.closedSteps else whenFalse.closedSteps
+  def stepsValue :List[WorkflowNodeId] =
+    if(cond().getOrElse(false)) whenTrue.stepsValue else whenFalse.stepsValue
+  def findNextSteps(currentUserRights: Seq[String], currentStep: WorkflowNodeId, isCreator: Boolean) : WorkflowAction =
+    if(cond().getOrElse(false)) whenTrue.findNextSteps(currentUserRights, currentStep, isCreator) else whenFalse.findNextSteps(currentUserRights, currentStep, isCreator)
+  def findBackSteps(currentUserRights: Seq[String], currentStep: WorkflowNodeId, isCreator: Boolean) : Seq[(WorkflowNodeId,(ChangeRequestId,EventActor, Option[String]) => Box[WorkflowNodeId])] =
+    if(cond().getOrElse(false)) whenTrue.findBackSteps(currentUserRights, currentStep, isCreator) else whenFalse.findBackSteps(currentUserRights, currentStep, isCreator)
+  def findStep(changeRequestId: ChangeRequestId) : Box[WorkflowNodeId] =
+    if(cond().getOrElse(false)) whenTrue.findStep(changeRequestId) else whenFalse.findStep(changeRequestId)
+  def getAllChangeRequestsStep() : Box[Map[ChangeRequestId,WorkflowNodeId]] =
+    if(cond().getOrElse(false)) whenTrue.getAllChangeRequestsStep else whenFalse.getAllChangeRequestsStep
+  def isEditable(currentUserRights: Seq[String], currentStep: WorkflowNodeId, isCreator: Boolean): Boolean =
+    if(cond().getOrElse(false)) whenTrue.isEditable(currentUserRights, currentStep, isCreator) else whenFalse.isEditable(currentUserRights, currentStep, isCreator)
+  def isPending(currentStep:WorkflowNodeId): Boolean =
+    if(cond().getOrElse(false)) whenTrue.isPending(currentStep) else whenFalse.isPending(currentStep)
+}
+
+
+trait ValidationNeeded {
+
+  // check is the given change request, initiated by actor, need to be validated or can
+  // be auto deployed.
+  def check(changeRequestId: ChangeRequestId, actor: EventActor): Box[Boolean]
+}
+
+class KonamiValidationNeeded(
+    roChangeRequestRepository: RoChangeRequestRepository
+  , changeRequestChangesSerialisation: ChangeRequestChangesSerialisation
+) extends ValidationNeeded {
+  val code = "↑↑↓↓←→←→BA"
+  override def check(changeRequestId: ChangeRequestId, actor: EventActor): Box[Boolean] = {
+    for {
+      opt <- roChangeRequestRepository.get(changeRequestId)
+    } yield {
+      val xml = opt match {
+                case Some(cr) => changeRequestChangesSerialisation.serialise(cr)
+                case None     => Full(NodeSeq.Empty)
+              }
+      if(xml.toString().contains(code)) {
```

Comment:
Can we ckeck we are in production or in dev run mode of Lift ? to prevent enabling it by default on production :)

Reply:
Actually, we don't really want to keep it in the released version :)